kratos/processes/simple_mortar_mapper_process.cpp

```diff
@@ -103,10 +145,118 @@ void SimpleMortarMapperProcess<TDim, TNumNodes, TVarType, THist>:: Execute()
 /***********************************************************************************/
 /***********************************************************************************/
 
-template< int TDim, int TNumNodes, class TVarType, HistoricalValues THist>
-void SimpleMortarMapperProcess<TDim, TNumNodes, TVarType, THist>::ResetNodalArea()
+template< int TDim, int TNumNodes, class TVarType, HistoricalValues THistOrigin, HistoricalValues THistDestination>
+void SimpleMortarMapperProcess<TDim, TNumNodes, TVarType, THistOrigin, THistDestination>::CheckAndPerformSearch()
 {
-    NodesArrayType& nodes_array = mrThisModelPart.Nodes();
+    // First we check if search already exists
+    bool search_exists = true;
+    // Iterate in the conditions
+    ConditionsArrayType& destination_conditions_array = mrDestinationModelPart.Conditions();
+    for(std::size_t i = 0; i < destination_conditions_array.size(); ++i) {
+        auto it_cond = destination_conditions_array.begin() + i;
+
+        if (!it_cond->Has( INDEX_SET )) {
+            search_exists = false; 
+            break;
+        }
+    }
+        
+    // Now we perform the corresponding search
+    if (search_exists == false) {
+        
+        // We create the variable INDEX_SET
+        #pragma omp parallel for 
+        for(int i = 0; i < static_cast<int>(destination_conditions_array.size()); ++i) {
+            auto it_cond = destination_conditions_array.begin() + i;
+            if (it_cond->Has(INDEX_SET) == false) 
+                it_cond->SetValue(INDEX_SET, Kratos::make_shared<IndexSet>()); 
+        }
+        
+        // A list that contents the all the points (from nodes) from the modelpart 
+        PointVector point_list_destination; 
+        
+        point_list_destination.clear();
+        
+        // Iterate in the conditions
+        ConditionsArrayType& origin_conditions_array = mrOriginModelPart.Conditions();
+        
+        // Creating a buffer for parallel vector fill
+        const int num_threads = OpenMPUtils::GetNumThreads();
+        std::vector<PointVector> points_buffer(num_threads);
+
+        #pragma omp parallel
+        {
+            const int thread_id = OpenMPUtils::ThisThread();
```

Comment:
i would propose to create a 
      
      PointVector points_buffer

here inside the parallel region. this way you will have one per thread

Reply:
OK

Reply:
I think the current way is the correct one, here the vector is not used, just filled in parallel

Reply:
The reason of my suggestion is to avoid using thread_id whenever possible. Call this a future oriented change